src/plugins/auto/utils/config.hpp

```diff
@@ -175,9 +150,32 @@ struct PluginConfig {
                 _passThroughConfig.emplace(kvp.first, kvp.second);
             } else if (kvp.first.find("AUTO_") == 0) {
                 _passThroughConfig.emplace(kvp.first, kvp.second);
-            } else if (kvp.first == ov::cache_dir.name()) {
-                _cacheDir = kvp.second;
-                _isSetCacheDir = true;
+            } else if (core_config.find(kvp.first) != core_config.end()) {
```

Comment:
why do we have to check here if cache_dir/batch is in core_config support keys?

Reply:
Because UpdateFromMap will be called by plugin.set_property() and plugin.compile_model(), but there will be different behavior:
    1) plugin.set_property() will throw exception if it received core properties, and pass core_config={} to trigger exception.
    2) plugin.compile_model() should receive and will handle core properties, and pass actual core_config to make it can be handled.